src/utilities/createRouteMethods.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { describe, expect, test } from 'vitest'
+import { Routes } from '@/types'
+import { createRouteMethods, resolveRoutes } from '@/utilities'
+
+const component = { template: '<div>This is component</div>' }
+
+describe('createRouteMethods', () => {
+  test.each([
+    [undefined],
+    [true],
+  ])('given route is named and is public, makes parent callable', () => {
+    const routes = [
+      {
+        name: 'parent',
+        path: '/parent',
+        component,
+      },
+    ] as const satisfies Routes
+    const resolved = resolveRoutes(routes)
+
+    const response = createRouteMethods(resolved)
+
+    expect(response.parent).toBeTypeOf('function')
+  })
+
+  test('given route is NOT public, returns empty object', () => {
+    const routes = [
+      {
+        name: 'parent',
+        path: '/parent',
+        public: false,
+        component,
+      },
+    ] as const satisfies Routes
+    const resolved = resolveRoutes(routes)
+
+    const response = createRouteMethods(resolved)
+
+    expect(response).toMatchObject({})
+  })
+
+  test.each([
+    [undefined],
+    [true],
+    [false],
+  ])('given parent route with named children, has property for child name', (isPublic) => {
+    const routes = [
+      {
+        name: 'parent',
+        path: '/parent',
+        children: [
+          {
+            name: 'child',
+            path: '/child',
+            public: isPublic,
+            component,
+          },
+        ],
+      },
+    ] as const satisfies Routes
+    const resolved = resolveRoutes(routes)
+
+    const response = createRouteMethods(resolved)
+
+    expect(response.parent.child).not.toBeUndefined()
```

Comment:
current implementation creates an empty object for each parent, which I don't think is right actually because if grandparent is public, parent is private, child is private, grandchild is public the currently implementation would make you go `response.grandparent.parent.child.grandchild()`, when I think our desired outcome is `response.grandparent.grandchild()`. Currently routes being public only really effects if the node is callable, so I'll need to update the logic slightly.

regardless, I wouldn't expect `response` to be empty because `parent` is public so `response.parent()` will be there

Reply:
I believe the way the types work is if a route has a name it's always part of the dot notation to get to children. So in grandparent, parent, child if parent has a name but is public false you still need to do grandparent.parent.child to call the child route. Open to changing that but I thought that's what we decided when we introduced public. Otherwise you'd just remove the name from the parent if you wanted to do grandparent.child